File: (5 kyu) Human Readable Time.js
/**
 * Human Readable Time (5 kyu)
 * https://www.codewars.com/kata/human-readable-time
 * Write a function, which takes a non-negative integer (seconds) as input
 * and returns the time in a human-readable format (HH:MM:SS)
 * HH = hours, padded to 2 digits, range: 00 - 99
 * MM = minutes, padded to 2 digits, range: 00 - 59
 * SS = seconds, padded to 2 digits, range: 00 - 59
 * The maximum time never exceeds 359999 (99:59:59)
 * You can find some examples in the test fixtures.
 */
const humanReadable = seconds =>
  [
    ('0' + Math.floor(seconds / 3600)).slice(-2),
    ('0' + Math.floor(seconds % 3600 / 60)).slice(-2),
    ('0' + seconds % 60).slice(-2)
  ].join(':');

File: 5kyu/(5 kyu) Human Readable Time/(5 kyu) Human Readable Time.py
# #1
#def make_readable(seconds):
#    hours = seconds/60**2
#    minutes = (seconds%60**2)/60
#    seconds = (seconds%60**2%60)
#    return "%02d:%02d:%02d" % (hours, minutes, seconds)

# #2
#def make_readable(seconds):
#    hours, seconds = divmod(seconds, 60 ** 2)
#    minutes, seconds = divmod(seconds, 60)
#    return '{:02}:{:02}:{:02}'.format(hours, minutes, seconds)

# #3
def make_readable(seconds):
    return '{:02}:{:02}:{:02}'.format(seconds / 3600, seconds / 60 % 60, seconds % 60)
